jenkins/bootstrap.py

```diff
@@ -215,7 +215,15 @@ def checkout(call, repo, branch, pull, ssh=False):
         refs, checkouts = branch_ref(branch)
 
     git = 'git'
-    call([git, 'init', repo])
+    if git_cache:
```

Comment:
The reason being that smaller tests are better than larger ones, and all the complexity lies in checkout.